edb/schema/pointers.py

```diff
@@ -1335,6 +1351,130 @@ def _alter_begin(
             self._validate_pointer_def(schema, context)
         return schema
 
+    def _alter_finalize(
+        self,
+        schema: s_schema.Schema,
+        context: sd.CommandContext,
+    ) -> s_schema.Schema:
+        schema = super()._alter_finalize(schema, context)  # type: ignore
+        self._validate_computables(schema, context)
+        return schema
+
+    def _validate_computables(
+        self,
+        schema: s_schema.Schema,
+        context: sd.CommandContext
+    ) -> None:
+        scls = self.scls
+
+        if scls.get_from_alias(schema):
+            return
+
+        is_computable = scls.is_pure_computable(schema)
+        is_owned = scls.get_owned(schema)
+        # Get the non-generic, explicitly declared ancestors as the
+        # limitations on computables apply to explicitly declared
+        # pointers, not just a long chain of inherited ones.
+        #
+        # Because this is potentially nested inside a command to
```

Comment:
What is the case where this can come up? If the ancestor has been deleted and so is this object, maybe there is a more straightforward way to just skip all this logic? If the ancestor has been deleted and this object is being rebased, then it might be a bug elsewhere that the rebase hasn't happened first, though I'm not totally sure.

Reply:
This is definitely masking a bug.

Reply:
Aha, it's because when a property is deleted, it first deletes its constraints (which will generate an alter of the child prop to delete *its* constraint) and *then* its children. (And that pattern shows up in cases where it doesn't need to because `ordering` generates wild and crazy diffs when inheritance is involved.) Maybe a cleaner expedient fix would be to skip validating computables unless something potentially relevant to computables changed (which might be a slight perf advantage too), but it's your call.

I found another bug while investigating this:
From 
```
abstract type Text { property x -> str { constraint exclusive } }         
abstract type Owned { property x -> str { constraint exclusive } }        
type Comment extending Text, Owned;
```
running `ALTER TYPE Text DROP PROPERTY x` gets us a fun `AttributeError: 'AlterProperty' object has no attribute 'scls'` crash. I'm going to follow up on that, which might end up cleaning up this whole snag.